### PR TITLE
Refactor FXIOS-5632 [v111] Remove os_log

### PR DESCRIPTION
**Pull request by OrlaM**
[FXIOS-5632](https://mozilla-hub.atlassian.net/browse/FXIOS-5632)
#13011

### CODE REVIEW

File: Extensions/NotificationService/NotificationService.swift [hunk 1]
```diff
@@ -8,23 +8,24 @@ import Shared
...
+    }
 
     // This is run when an APNS notification with `mutable-content` is received.
     // If the app is backgrounded, then the alert notification is displayed.
     // If the app is foregrounded, then the notification.userInfo is passed straight to
     // AppDelegate.application(_:didReceiveRemoteNotification:completionHandler:)
     // Once the notification is tapped, then the same userInfo is passed to the same method in the AppDelegate.
     override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
-        consoleLog("push received")
+        logger.log("push notification received", level: .info, category: .lifecycle)
```
Comment (lmarceau) [inline Extensions/NotificationService/NotificationService.swift:28]: Oh, I had not realized we could use .lifecycle as such. Good point. I think in my other PR removing .core we could replace some with that category as well. Let me know what you think.

**Review by lmarceau — APPROVED** [review]

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Project coverage: 29.25%</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Edited 2 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 0 files</td>
    </tr>
  </tbody>
</table>

## NotificationService.appex: Coverage: 20.28
| File | Coverage ||
| --- | --- | --- |
NotificationService.swift | 0.0% | ⚠️

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against f5cf814f5db8f534e7e89bd46ba8c66ee35c72ad
</p>

File: Extensions/NotificationService/NotificationService.swift (hunk 1 above)
Comment (OrlaM) [reply Extensions/NotificationService/NotificationService.swift:28]: Yeah that could work for some of them. I think we might also want to add a general category too, I feel there will be many cases where people don't know which category to use so will misuse them. Better to have them end up in a default or unlabelled category than to add unwanted noise to the others.

**Comment by OrlaM** [comment]
BR passed everything right up until it aborted at the last second. Based on what Isabel said about this issue on BR I think it's safe to merge without re-running it.

File: Extensions/NotificationService/NotificationService.swift (hunk 1 above)
Comment (lmarceau) [reply Extensions/NotificationService/NotificationService.swift:28]: Ok I'll add another task for this, good point. `Unlabeled` make sense, as its less enticing to use than `General` (which could refrain people from using another category the log would fit in).

**Pull request merged**